src/theory/strings/inference_manager.cpp

```diff
@@ -186,13 +186,33 @@ void InferenceManager::sendInference(const std::vector<Node>& exp,
   sendInference(exp, exp_n, eq, c, asLemma);
 }
 
-void InferenceManager::sendInference(const InferInfo& i)
+void InferenceManager::sendInference(const std::vector<Node>& exp,
+                                     const std::vector<Node>& exp_n,
+                                     Node eq,
+                                     Inference infer,
+                                     bool asLemma)
+{
+  d_parent.getStatistics()->d_inferences << infer;
```

Comment:
Actually I'd like to move away from using parent, can we pass a reference to stats and then use it directly (similar to how we pass other utilities like EqualityEngine/SolverState/etc.?).

Reply:
Sounds good, done.